torch/distributed/hooks.py

```diff
@@ -0,0 +1,138 @@
+import logging
```

Comment:
yea, this is a good point.  i think we can answer it in 2 parts.

first is that doing a background-thread in python here is more of a demonstration iiuc.  we'd suggest a more robust way to use these hooks is to have them initiate a fast, async request to a service to store some data and then return.  this way, all the critical functionality is happening in the remote service (in another process, or even machine)

second is that we could also support registering c++ hooks.  i think it would be trivial to add, following exactly from the python API and mirroring the 'CollectiveStatus' object as a c++ class.  We might do this but if you want it and want to send a PR that'd be great too.